File: app/src/main/java/com/meharenterprises/originsms/receivers/SendStatusReceiver.java
package com.meharenterprises.originsms.receivers;

import android.content.BroadcastReceiver;
import android.content.ContentValues;
import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.provider.Telephony;
import android.widget.Toast;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.exifinterface.media.ExifInterface;
import com.meharenterprises.originsms.R;
import kotlin.Metadata;
import kotlin.Unit;
import kotlin.io.CloseableKt;
import kotlin.jvm.internal.Intrinsics;

/* compiled from: SendStatusReceiver.kt */
@Metadata(d1 = {"\u0000 \n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\u0018\u00002\u00020\u0001B\u0005¢\u0006\u0002\u0010\u0002J\u0010\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u0006H\u0002J\u0018\u0010\u0007\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u00062\u0006\u0010\b\u001a\u00020\tH\u0016¨\u0006\n"}, d2 = {"Lcom/meharenterprises/originsms/receivers/SendStatusReceiver;", "Landroid/content/BroadcastReceiver;", "()V", "markMostRecentSentMessageFailed", "", "context", "Landroid/content/Context;", "onReceive", "intent", "Landroid/content/Intent;", "app_debug"}, k = 1, mv = {1, 9, 0}, xi = ConstraintLayout.LayoutParams.Table.LAYOUT_CONSTRAINT_VERTICAL_CHAINSTYLE)
/* loaded from: classes4.dex */
public final class SendStatusReceiver extends BroadcastReceiver {
    @Override // android.content.BroadcastReceiver
    public void onReceive(Context context, Intent intent) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(intent, "intent");
        switch (getResultCode()) {
            case -1:
            case 0:
            default:
                return;
            case 1:
            case 2:
            case 3:
            case 4:
                Toast.makeText(context, context.getString(R.string.status_failed), 0).show();
                markMostRecentSentMessageFailed(context);
                return;
        }
    }

    private final void markMostRecentSentMessageFailed(Context context) {
        int idIdx;
        Cursor query = context.getContentResolver().query(Telephony.Sms.CONTENT_URI, new String[]{"_id"}, "type = ?", new String[]{ExifInterface.GPS_MEASUREMENT_2D}, "date DESC LIMIT 1");
        if (query != null) {
            Cursor cursor = query;
            try {
                Cursor cursor2 = cursor;
                if (cursor2.moveToFirst() && (idIdx = cursor2.getColumnIndex("_id")) >= 0) {
                    long messageId = cursor2.getLong(idIdx);
                    ContentValues values = new ContentValues();
                    values.put("type", (Integer) 5);
                    context.getContentResolver().update(Telephony.Sms.CONTENT_URI, values, "_id = ?", new String[]{String.valueOf(messageId)});
                }
                Unit unit = Unit.INSTANCE;
                CloseableKt.closeFinally(cursor, null);
            } catch (Throwable th) {
                try {
                    throw th;
                } catch (Throwable th2) {
                    CloseableKt.closeFinally(cursor, th);
                    throw th2;
                }
            }
        }
    }
}

File: app/src/main/java/com/meharenterprises/originsms/receivers/SendStatusReceiver.kt
package com.meharenterprises.originsms.receivers

import android.content.BroadcastReceiver
import android.content.ContentValues
import android.content.Context
import android.content.Intent
import android.provider.Telephony
import android.telephony.SmsManager
import android.widget.Toast
import com.meharenterprises.originsms.R

/**
 * Receives the "sent" PendingIntent callback from SmsManager (for both SMS
 * and MMS sends — MMS routes its sentIntent through this same receiver).
 * On failure, the most recent SENT-box row for this conversation is flagged
 * as FAILED in the system provider so ThreadActivity's "Not sent. Tap to
 * retry." status renders correctly instead of staying stuck on "Sent".
 */
class SendStatusReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        when (resultCode) {
            android.app.Activity.RESULT_OK -> {
                // Sent successfully — the optimistic SENT-box row inserted at
                // send time already reflects the correct final state.
            }
            SmsManager.RESULT_ERROR_GENERIC_FAILURE,
            SmsManager.RESULT_ERROR_NO_SERVICE,
            SmsManager.RESULT_ERROR_NULL_PDU,
            SmsManager.RESULT_ERROR_RADIO_OFF -> {
                Toast.makeText(context, context.getString(R.string.status_failed), Toast.LENGTH_SHORT).show()
                markMostRecentSentMessageFailed(context)
            }
        }
    }

    private fun markMostRecentSentMessageFailed(context: Context) {
        context.contentResolver.query(
            Telephony.Sms.CONTENT_URI,
            arrayOf(Telephony.Sms._ID),
            "${Telephony.Sms.TYPE} = ?",
            arrayOf(Telephony.Sms.MESSAGE_TYPE_SENT.toString()),
            "${Telephony.Sms.DATE} DESC LIMIT 1"
        )?.use { cursor ->
            if (cursor.moveToFirst()) {
                val idIdx = cursor.getColumnIndex(Telephony.Sms._ID)
                if (idIdx >= 0) {
                    val messageId = cursor.getLong(idIdx)
                    val values = ContentValues().apply {
                        put(Telephony.Sms.TYPE, Telephony.Sms.MESSAGE_TYPE_FAILED)
                    }
                    context.contentResolver.update(
                        Telephony.Sms.CONTENT_URI,
                        values,
                        "${Telephony.Sms._ID} = ?",
                        arrayOf(messageId.toString())
                    )
                }
            }
        }
    }
}

/**
 * Receives delivery confirmation broadcasts (when the carrier confirms the
 * recipient's device received the message). Delivery reports are
 * best-effort and carrier-dependent; when one arrives it simply confirms
 * what the UI already shows as "Sent" — no provider write is needed since
 * this app does not persist a separate delivered/undelivered column.
 */
class DeliveryStatusReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        // No-op by design: absence of a delivery report is normal on many
        // carriers/recipient devices, so the UI intentionally does not treat
        // "no report received" as a failure state.
    }
}
